client/delegations.go

```diff
@@ -0,0 +1,202 @@
+package client
+
+import (
+	"github.com/theupdateframework/go-tuf/data"
+	"github.com/theupdateframework/go-tuf/verify"
+)
+
+// getTargetFileMeta searches for a verified TargetFileMeta matching a file name
+// Requires a local snapshot to be loaded and is locked to the snapshot versions.
+// Searches through delegated targets following TUF spec 1.0.19 section 5.6.
+func (c *Client) getTargetFileMeta(file string) (data.TargetFileMeta, error) {
+	snapshot, err := c.loadLocalSnapshot()
+	if err != nil {
+		return data.TargetFileMeta{}, err
+	}
+
+	// verifiers is map of parent targets name to an associated DelegationsVerifier
+	// that can verify all child targets pointed by delegatedRoles in the parent targets
+	verifiers := make(map[string]verify.DelegationsVerifier)
```

Comment:
Note to self: think about how the same role can be given a different threshold of keys in different delegations.

Example: imagine A delegates to B and C, and both B and C delegate to D. However, B and C use different keys to delegate to D.

Like Ethan said, consider not having any variable that persists across iterations. Easier to understand and not make mistakes.